### PR TITLE
Replace `.npmignores` with the `file` field

### PR DESCRIPTION
**Pull request by sdadn**
### Summary

This PR replaces `.npmignore` files with the `files` prop to denote what files will be published. The `.npmignore` works by blacklisting files, while `files` works by whitelisting files to be published. This change was made as it is more convenient to keep track of files that need to be published than finding potential files that don't need to be published. It also results in fewer issues when running `npm pack` locally.

### Testing

This change was tested using:

- [ ] WDIO
- [ ] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [x] Other (please describe below)
- [ ] No tests are needed

This was tested by running `npm pack` on `terra-clinical-header` and verifying that all necessary files are included:

CleanShot 2024-02-15 at 11 34 28

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [ ] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-XXXX 

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

**Comment by sdadn** [comment]
> Fails
> 🚫	Please include a CHANGELOG entry for each changed package this PR. Looks like a CHANGELOG entry is missing for:
> * terra-action-footer
> * terra-action-header
> * terra-alert
> * terra-arrange
> * terra-avatar
> * terra-badge
> * terra-base
> * terra-breakpoints
> * terra-button-group
> * terra-button
> * terra-card
> * terra-cell-grid
> * terra-content-container
> * terra-core-docs
> * terra-demographics-banner
> * terra-dialog
> * terra-divider
> * terra-docs
> * terra-dropdown-button
> * terra-dynamic-grid
> * terra-form-checkbox
> * terra-form-field
> * terra-form-fieldset
> * terra-form-input
> * terra-form-radio
> * terra-form-select
> * terra-form-textarea
> * terra-grid
> * terra-heading
> * terra-html-table
> * terra-hyperlink
> * terra-i18n
> * terra-icon
> * terra-image
> * terra-list
> * terra-mixins
> * terra-outline-view-table
> * terra-overlay
> * terra-paginator
> * terra-profile-image
> * terra-progress-bar
> * terra-responsive-element
> * terra-scroll
> * terra-search-field
> * terra-section-header
> * terra-show-hide
> * terra-signature
> * terra-spacer
> * terra-status-view
> * terra-status
> * terra-switch
> * terra-tag
> * terra-text
> * terra-toggle-button
> * terra-toggle-section-header
> * terra-toggle
> * terra-toolbar
> * terra-visually-hidden-text
> 
> Generated by 🚫 [dangerJS](https://danger.systems/js) against [c86e64e](https://github.com/cerner/terra-core/commit/c86e64e825cecdaded87a1cec652a52ea4125449)

CHANGELOG updates are not needed.

File: packages/terra-action-footer/package.json [hunk 1]
```diff
@@ -42,5 +42,13 @@
     "wdio-lowlight": "cd ../.. && terra wdio --themes clinical-lowlight-theme",
     "wdio-fusion": "cd ../.. && terra wdio --themes orion-fusion-theme",
     "wdio": "npm run wdio-default && npm run wdio-lowlight && npm run wdio-fusion"
-  }
+  },
+  "files": [
```
Comment (nikhitasharma) [inline packages/terra-action-footer/package.json:46]: Should translations be in some of these lists.

File: packages/terra-toggle/package.json [hunk 2]
```diff
@@ -43,5 +43,13 @@
     "wdio-lowlight": "cd ../.. && terra wdio --themes clinical-lowlight-theme",
     "wdio-fusion": "cd ../.. && terra wdio --themes orion-fusion-theme",
     "wdio": "npm run wdio-default && npm run wdio-lowlight && npm run wdio-fusion"
-  }
+  },
+  "files": [
```
Comment (nikhitasharma) [inline packages/terra-toggle/package.json:47]: There is a docs folder under this component and some others as well with Readme and upgrade guides. Should those be included as well?

File: packages/terra-button-group/package.json [hunk 3]
```diff
@@ -44,5 +44,13 @@
     "wdio-lowlight": "cd ../.. && terra wdio --themes clinical-lowlight-theme",
     "wdio-fusion": "cd ../.. && terra wdio --themes orion-fusion-theme",
     "wdio": "npm run wdio-default && npm run wdio-lowlight && npm run wdio-fusion"
-  }
+  },
+  "files": [
```
Comment (nikhitasharma) [inline packages/terra-button-group/package.json:48]: If translations are added to any of the components in future, should those be added in package.json as well for them to work in terra ui?

Comment (sdadn) [reply packages/terra-button-group/package.json:48]: Yes, if any additional folders get added in the future, they will need to be added to this list.

File: packages/terra-toggle/package.json (hunk 2 above)
Comment (sdadn) [reply packages/terra-toggle/package.json:47]: That's a good catch! Looks like those files already exist in terra-core-docs so they can be excluded from here:

https://github.com/cerner/terra-core/tree/main/packages/terra-core-docs/src/terra-dev-site/doc/toggle

**Review by nikhitasharma — APPROVED** [review]

**Pull request merged**